### ISSUE TITLE
DifferentiationInterface instead of individual backends?

### ISSUE THREAD

**Issue by gdalle** [SciML/Integrals.jl#258]
Following our conversation with @lxvm today, opening this issue to keep track of a potential DI integration here (pun intended). What would be the major hurdles?

**Comment by ChrisRackauckas** [comment]
I don't think DI has a rules system that applies rules to all backends?

**Comment by gdalle** [comment]
No, for that we would still borrow individual rule systems. But inside of these rules we compute derivatives, and it can be with an arbitrary backend, not only the one we define the rule inside

**Comment by ChrisRackauckas** [comment]
I see, yes that part makes sense. Yes it doesn't need the SciMLSensitivity complexity because the differentiation here is pretty trivial, it just needs an easy way to pass what AD to use internally.
